File: Answer/3754.连接非零数字并乘以其数字和-i.ts
/*
 * @lc app=leetcode.cn id=3754 lang=typescript
 *
 * [3754] 连接非零数字并乘以其数字和 I
 */

// @lc code=start
function sumAndMultiply(n: number): number {
    let x = 0;
    let pow = 1;
    let sum = 0;
    while (n > 0) {
        const d = n % 10;
        if (d !== 0) {
            x += d * pow;
            pow *= 10;
            sum += d;
        }
        n = Math.floor(n / 10);
    }
    return x * sum;
}
// @lc code=end

File: Answer/3754.连接非零数字并乘以其数字和-i.go
/*
 * @lc app=leetcode.cn id=3754 lang=golang
 *
 * [3754] 连接非零数字并乘以其数字和 I
 */

// @lc code=start
func sumAndMultiply(n int) int64 {
	x := 0
	pow := 1
	sum := 0
	for n > 0 {
		d := n % 10
		if d != 0 {
			x += d * pow
			pow *= 10
			sum += d
		}
		n /= 10
	}
	return int64(x) * int64(sum)
}

// @lc code=end
